x-pack/legacy/plugins/alerting/server/init.ts

```diff
@@ -47,138 +26,45 @@ interface Server extends Legacy.Server {
-  const { callWithRequest } = server.plugins.elasticsearch.getCluster('admin');
-  const spaces = createOptionalPlugin<SpacesPlugin>(
-    config,
-    'xpack.spaces',
-    server.plugins,
-    'spaces'
-  );
-  const security = createOptionalPlugin<SecurityPluginSetupContract>(
-    config,
-    'xpack.security',
-    kbnServer.newPlatform.setup.plugins,
-    'security'
-  );
-
-  server.plugins.xpack_main.registerFeature({
-    id: 'alerting',
-    name: 'Alerting',
-    app: ['alerting', 'kibana'],
-    privileges: {
-      all: {
-        savedObject: {
-          all: ['alert'],
-          read: [],
-        },
-        ui: [],
-        api: ['alerting-read', 'alerting-all'],
-      },
-      read: {
-        savedObject: {
-          all: [],
-          read: ['alert'],
-        },
-        ui: [],
-        api: ['alerting-read'],
-      },
+  const newPlatform = ((server as unknown) as KbnServer).newPlatform;
+  const coreSetup = {
+    elasticsearch: server.plugins.elasticsearch,
+    savedObjects: server.savedObjects,
+    http: {
+      route: server.route.bind(server),
+      basePath: newPlatform.setup.core.http.basePath,
     },
-  });
-
-  // Encrypted attributes
-  server.plugins.encrypted_saved_objects.registerType({
-    type: 'alert',
-    attributesToEncrypt: new Set(['apiKey']),
-    attributesToExcludeFromAAD: new Set([
-      'scheduledTaskId',
-      'muted',
-      'mutedInstanceIds',
-      'updatedBy',
-    ]),
-  });
-
-  function getServices(request: any): Services {
-    return {
-      log: (...args) => server.log(...args),
-      callCluster: (...args) => callWithRequest(request, ...args),
-      savedObjectsClient: server.savedObjects.getScopedSavedObjectsClient(request),
-    };
-  }
-  function getBasePath(spaceId?: string): string {
-    return spaces.isEnabled && spaceId
-      ? spaces.getBasePath(spaceId)
-      : ((server.config().get('server.basePath') || '') as string);
-  }
-  function spaceIdToNamespace(spaceId?: string): string | undefined {
-    return spaces.isEnabled && spaceId ? spaces.spaceIdToNamespace(spaceId) : undefined;
-  }
+  };
+  const pluginsSetup = {
+    security: newPlatform.setup.plugins.security as SecurityPluginSetupContract | undefined,
+    task_manager: server.plugins.task_manager,
+    // TODO: Currently a function because it's an optional dependency that
+    // initializes after this function is called
+    spaces: () => server.plugins.spaces,
+    actions: server.plugins.actions,
+    xpack_main: server.plugins.xpack_main,
+    encrypted_saved_objects: server.plugins.encrypted_saved_objects,
+  };
 
-  const alertTypeRegistry = new AlertTypeRegistry({
-    getServices,
-    isSecurityEnabled: security.isEnabled,
-    taskManager,
-    executeAction: server.plugins.actions.execute,
-    encryptedSavedObjectsPlugin: server.plugins.encrypted_saved_objects,
-    getBasePath,
-    spaceIdToNamespace,
+  const plugin = new Plugin({
+    logger: {
```

Comment:
Since you're already compatible with the new API, you could just use the NP logger here `kbnServer.newPlatform.coreContext.logger`

Reply:
Awesome! Fixed in https://github.com/elastic/kibana/pull/46954/commits/68a6bcd25ec3a67e555d9111e554011abed678fe.